sql/planbuilder/cte.go

```diff
@@ -84,22 +84,26 @@ func (b *Builder) buildRecursiveCte(inScope *scope, union *ast.Union, name strin
 	l, r := splitRecursiveCteUnion(name, union)
 	if r == nil {
 		// not recursive
-		cteScope := b.buildSelectStmt(inScope, union)
+		sqScope := inScope.push()
+		sqScope.initSubquery()
```

Comment:
This is probably better expressed as `inScope.pushSubquery()`

There's basically no use case for calling this except immediatley after construction, so move it to a constructor

Reply:
the case is your next comment, i added a method to do both but left the init for cases where we want replace() not push()